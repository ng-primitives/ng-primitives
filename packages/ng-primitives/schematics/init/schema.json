{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "ng-primitives-init",
  "title": "Angular Primitives ng-add schematic",
  "type": "object",
  "properties": {},
  "required": []
}
