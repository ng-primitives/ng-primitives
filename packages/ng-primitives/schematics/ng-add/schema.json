{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "angular-primitives-ng-add",
  "title": "Angular Primitives ng-add",
  "type": "object",
  "properties": {},
  "required": []
}
