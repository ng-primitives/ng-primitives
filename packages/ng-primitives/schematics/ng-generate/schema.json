{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SchematicsAngularComponent",
  "title": "Angular Primitives Component Options Schema",
  "type": "object",
  "description": "Create a reusable Angular Primitives component.",
  "additionalProperties": false,
  "properties": {
    "primitive": {
      "type": "string",
      "description": "The name of the primitive to generate.",
      "enum": [
        "accordion",
        "avatar",
        "button",
        "checkbox",
        "date-picker",
        "dialog",
        "file-upload",
        "form-field",
        "input",
        "listbox",
        "pagination",
        "progress",
        "radio",
        "search",
        "separator",
        "slider",
        "switch",
        "tabs",
        "textarea",
        "toast",
        "toggle",
        "toggle-group",
        "toolbar"
      ],
      "x-prompt": "What primitive would you like to generate?",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path where the component files should be created, relative to the current workspace.",
      "x-prompt": "What is the path to create the component?"
    },
    "prefix": {
      "type": "string",
      "description": "A prefix to be added to the component's selector. For example, if the prefix is `app` and the component name is `button`, the selector will be `app-button`.",
      "alias": "p",
      "oneOf": [
        {
          "maxLength": 0
        },
        {
          "minLength": 1,
          "format": "html-selector"
        }
      ]
    },
    "componentSuffix": {
      "type": "string",
      "description": "Append the specified suffix to the component name. If the component name is `Button` and the suffix is `Component`, the generated component will be named `ButtonComponent`.",
      "default": "Component"
    },
    "fileSuffix": {
      "type": "string",
      "default": "component",
      "description": "Generate component template files with the specified suffix."
    },
    "exampleStyles": {
      "type": "boolean",
      "description": "Whether example styles should be generated.",
      "default": true
    }
  },
  "required": ["primitive", "path"]
}
