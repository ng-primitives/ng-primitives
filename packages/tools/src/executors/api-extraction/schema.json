{
  "$schema": "https://json-schema.org/schema",
  "version": 2,
  "title": "ApiExtraction executor",
  "description": "",
  "type": "object",
  "properties": {},
  "required": []
}
