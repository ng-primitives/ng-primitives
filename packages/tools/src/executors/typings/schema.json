{
  "$schema": "https://json-schema.org/schema",
  "version": 2,
  "title": "Typings executor",
  "description": "",
  "type": "object",
  "properties": {},
  "required": []
}
