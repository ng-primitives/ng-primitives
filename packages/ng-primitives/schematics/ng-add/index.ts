import { Rule, Tree } from '@angular-devkit/schematics';
import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
import {
  addPackageJsonDependency,
  NodeDependencyType,
} from '@schematics/angular/utility/dependencies';

export default function ngAdd(): Rule {
  return (tree: Tree, context) => {
    addPackageJsonDependency(tree, {
      type: NodeDependencyType.Default,
      name: '@angular/cdk',
      version: '^19.0.0',
      overwrite: false,
    });

    addPackageJsonDependency(tree, {
      type: NodeDependencyType.Default,
      name: '@floating-ui/dom',
      version: '^1.6.0',
      overwrite: false,
    });

    context.addTask(new NodePackageInstallTask());
    return tree;
  };
}
