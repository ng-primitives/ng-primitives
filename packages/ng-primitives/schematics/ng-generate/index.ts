import { strings } from '@angular-devkit/core';
import {
  FileEntry,
  Rule,
  apply,
  forEach,
  mergeWith,
  move,
  template,
  url,
} from '@angular-devkit/schematics';
import { AngularPrimitivesComponentSchema } from './schema';

export default function generatePrimitive(options: AngularPrimitivesComponentSchema): Rule {
  return () => {
    const templateSource = apply(url(`./templates/${options.primitive}`), [
      template({
        ...options,
        ...strings,
        prefix: options.prefix ?? 'app',
        fileSuffix: options.fileSuffix ?? 'component',
      }),
      forEach((fileEntry: FileEntry) => {
        if (fileEntry.path.endsWith('.template')) {
          const newPath = fileEntry.path.replace('.template', '');
          return {
            path: newPath,
            content: fileEntry.content,
          } as FileEntry;
        }
        return fileEntry;
      }),
      move(options.path),
    ]);

    return mergeWith(templateSource);
  };
}
