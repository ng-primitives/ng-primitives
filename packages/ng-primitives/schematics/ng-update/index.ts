import { Rule, SchematicContext } from '@angular-devkit/schematics';
import {
  createMigrationSchematicRule,
  NullableDevkitMigration,
  TargetVersion,
} from '@angular/cdk/schematics';
import { NgpTargetVersion } from './data/changeset';
import { ngpUpgradeData } from './upgrade-data';

const migrations: NullableDevkitMigration[] = [];

/** Entry point for the migration schematics with target of Angular Primitives v0.24.0 */
export function updateToV0_24_0(): Rule {
  return createMigrationSchematicRule(
    // we have to cast this as the enum only allows Angular versions, but it doesn't matter for us
    NgpTargetVersion.V0_24_0 as unknown as TargetVersion,
    migrations,
    ngpUpgradeData,
    onMigrationComplete,
  );
}

/** Function that will be called when the migration completed. */
function onMigrationComplete(
  context: SchematicContext,
  targetVersion: TargetVersion,
  hasFailures: boolean,
) {
  context.logger.info('');
  context.logger.info(`  ✓  Updated Angular Primitives to ${targetVersion}`);
  context.logger.info('');

  if (hasFailures) {
    context.logger.warn(
      '  ⚠  Some issues were detected but could not be fixed automatically. Please check the ' +
        'output above and fix these issues manually.',
    );
  }
}
