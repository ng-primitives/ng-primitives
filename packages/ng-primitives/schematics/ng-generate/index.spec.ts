import { SchematicTestRunner, UnitTestTree } from '@angular-devkit/schematics/testing';
import { Schema as ApplicationOptions } from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import { AngularPrimitivesComponentSchema } from './schema';

describe('Component Schematic', () => {
  const schematicRunner = new SchematicTestRunner(
    'ng-primitives',
    require.resolve('../../collection.json'),
  );

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    newProjectRoot: 'projects',
    version: '6.0.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'bar',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    skipTests: false,
    skipPackageJson: false,
  };

  let appTree: UnitTestTree;

  beforeEach(async () => {
    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'workspace',
      workspaceOptions,
    );
    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'application',
      appOptions,
      appTree,
    );
  });

  it('should create a primitive with the correct file suffix', async () => {
    const options: AngularPrimitivesComponentSchema = {
      primitive: 'button',
      path: 'projects/bar/src/app/button',
      fileSuffix: 'ng',
    };

    const tree = await schematicRunner.runSchematic('primitive', options, appTree);
    expect(tree.files).toContain('/projects/bar/src/app/button/button.ng.ts');
  });

  it('should allow a primitive to be created with no file suffix', async () => {
    const options: AngularPrimitivesComponentSchema = {
      primitive: 'button',
      path: 'projects/bar/src/app/button',
      fileSuffix: '',
    };

    const tree = await schematicRunner.runSchematic('primitive', options, appTree);
    expect(tree.files).toContain('/projects/bar/src/app/button/button.ts');
  });

  it('should create a primitive with the correct prefix', async () => {
    const options: AngularPrimitivesComponentSchema = {
      primitive: 'button',
      path: 'projects/bar/src/app/button',
      prefix: 'foo',
    };

    const tree = await schematicRunner.runSchematic('primitive', options, appTree);
    expect(tree.files).toContain('/projects/bar/src/app/button/button.component.ts');
    const content = tree.readContent('/projects/bar/src/app/button/button.component.ts');
    expect(content).toContain("selector: 'button[foo-button]'");
  });

  it('should create a primitive with the correct component suffix', async () => {
    const options: AngularPrimitivesComponentSchema = {
      primitive: 'button',
      path: 'projects/bar/src/app/button',
      componentSuffix: '',
    };

    const tree = await schematicRunner.runSchematic('primitive', options, appTree);
    expect(tree.files).toContain('/projects/bar/src/app/button/button.component.ts');
    const content = tree.readContent('/projects/bar/src/app/button/button.component.ts');
    expect(content).toContain('export class Button {');
  });

  it('should create a button primitive', async () => {
    const options: AngularPrimitivesComponentSchema = {
      primitive: 'button',
      path: 'projects/bar/src/app/button',
    };

    const tree = await schematicRunner.runSchematic('primitive', options, appTree);
    expect(tree.files).toContain('/projects/bar/src/app/button/button.component.ts');
    const content = tree.readContent('/projects/bar/src/app/button/button.component.ts');
    expect(content).toMatchSnapshot();
  });

  it('should create an input primitive', async () => {
    const options: AngularPrimitivesComponentSchema = {
      primitive: 'input',
      path: 'projects/bar/src/app/input',
    };

    const tree = await schematicRunner.runSchematic('primitive', options, appTree);
    expect(tree.files).toContain('/projects/bar/src/app/input/input.component.ts');
    const content = tree.readContent('/projects/bar/src/app/input/input.component.ts');
    expect(content).toMatchSnapshot();
  });

  it('should create an accordion primitive', async () => {
    const options: AngularPrimitivesComponentSchema = {
      primitive: 'accordion',
      path: 'projects/bar/src/app/accordion',
    };

    const tree = await schematicRunner.runSchematic('primitive', options, appTree);
    expect(tree.files).toContain('/projects/bar/src/app/accordion/accordion.component.ts');
    const content = tree.readContent('/projects/bar/src/app/accordion/accordion.component.ts');
    expect(content).toMatchSnapshot();
  });
});
