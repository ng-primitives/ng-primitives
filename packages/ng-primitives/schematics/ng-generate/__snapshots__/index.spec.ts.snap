// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Schematic should create a button primitive 1`] = `
"import { Component } from '@angular/core';
import { NgpButton } from 'ng-primitives/button';

@Component({
  selector: 'button[app-button]',
  hostDirectives: [{ directive: NgpButton, inputs: ['disabled'] }],
  template: '<ng-content />',
  styles: \`
    :host {
    }

    :host[data-hover] {
    }

    :host[data-focus-visible] {
    }

    :host[data-press] {
    }
  \`,
})
export class ButtonComponent {}
"
`;

exports[`Component Schematic should create an accordion primitive 1`] = `
"import { Component } from '@angular/core';
import { NgpAccordion } from 'ng-primitives/accordion';

@Component({
  selector: 'app-accordion',
  hostDirectives: [
    {
      directive: NgpAccordion,
      inputs: [
        'ngpAccordionValue:value',
        'ngpAccordionType:type',
        'ngpAccordionCollapsible:collapsible',
        'ngpAccordionDisabled:disabled',
        'ngpAccordionOrientation:orientation',
      ],
    },
  ],
  template: \`
    <ng-content />
  \`,
  styles: \`
    :host {
      display: flex;
      flex-direction: column;
    }
  \`,
})
export class AccordionComponent {}
"
`;

exports[`Component Schematic should create an input primitive 1`] = `
"import { Component } from '@angular/core';
import { NgpInput } from 'ng-primitives/input';

@Component({
  selector: 'input[app-input]',
  hostDirectives: [{ directive: NgpInput, inputs: ['disabled'] }],
  template: '',
  styles: \`
    :host {
    }

    :host[data-focus] {
    }

    :host::placeholder {
    }
  \`,
})
export class InputComponent {}
"
`;
